File: sdk/python/src/bf_sdk/resources/snapshots.py
from __future__ import annotations

from bf_sdk._http import HttpClient
from bf_sdk.models.snapshot import Snapshot


class Snapshots:
    """Snapshot listing."""

    def __init__(self, http: HttpClient) -> None:
        self._http = http

    def list(self) -> list[Snapshot]:
        data = self._http.get("/api/v1/snapshots")
        return [Snapshot.model_validate(s) for s in data.get("snapshots", [])]

File: sdk/typescript/src/resources/snapshots.ts
import type { HttpClient } from "../http";
import { parseSnapshotListResponse, type Snapshot } from "../models/snapshot";

export class Snapshots {
  private readonly http: HttpClient;

  public constructor(http: HttpClient) {
    this.http = http;
  }

  public async list(): Promise<Snapshot[]> {
    const data = await this.http.get("/api/v1/snapshots");
    return parseSnapshotListResponse(data).snapshots;
  }
}
